lib/rouge/lexers/kotlin.rb

```diff
@@ -29,6 +29,8 @@ class Kotlin < RegexLexer
 
       id = %r'(#{name_backtick})'
 
+      plugin_name = %r'@?[_\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Nl}][\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Nl}\p{Nd}\p{Pc}\p{Cf}\p{Mn}\p{Mc}\p-]*'
```

Comment:
Two issues with this:

1. It seems duplicative of `name`. Perhaps extract the content of the character class into a string and then create both regular expressions using interpolation?

2. In Ruby 2.6.3 I get an error in `irb` when making a regular expression with `\p-`. I think it should just be `-` (I'd also move it to the front of the class, ahead of `_` so that it doesn't create ambiguity as to whether it's a range).

In light of the discussion in #1325, it may make sense to wait until that's addressed before making further changes to this PR.

Reply:
The \p was me still getting used to some of the Ruby notation.
In the end after making changes in the other PR to simplify the name regexes, this fix could be done just be allowing '-' as well as '_' in the name regex. This is valid, as users may have different naming conventions for their properties/functions etc.

For example, a common convention for test function names includes underscores.